packages/icon/cli/generate-components.js

```diff
@@ -7,23 +7,35 @@ const svgo = new SVGO({ plugins: [{ removeXMLNS: true }] })
 
 const regex = />/
 const componentTemplate = (svg, componentName, core) => `
-import React from 'react'
-import PropTypes from 'prop-types'
-import Icon from ${
-  core ? "'../index.js'" : "'@pluralsight/ps-design-system-icon'"
+import React, { HTMLAttributes } from 'react'
+import Icon from ${core ? "'../'" : "'@pluralsight/ps-design-system-icon'"}
+import { RefForwardingComponent } from '@pluralsight/ps-design-system-util'
+
+
+interface IconStatics {
+  sizes: typeof Icon.sizes
+  colors: typeof Icon.colors
 }
-export const ${componentName} =  React.forwardRef(function ${componentName}({'aria-label': ariaLabel, ...props}, ref){ return (
-  <Icon {...props} ref={ref}>
-    ${svg.replace(
-      regex,
-      "role='img' {...(ariaLabel && { 'aria-label': ariaLabel })}>"
-    )}
-  </Icon>
-)})
-${componentName}.displayName = '${componentName}'
-${componentName}.propTypes = {
-  'aria-label': PropTypes.string,
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
```

Comment:
Something like `interface ${capitalize(componentName)}Props` would be good here.